data/api_v1_to_v2.go

```diff
@@ -136,41 +143,60 @@ func (a *API) getDataV2(ctx context.Context, res *httpResponseWriter) error {
 	// Fetch data from store and V2 API (for cbg)
 	var wg sync.WaitGroup
 	sessionToken := res.Header.Get("x-tidepool-session-token")
-	wg.Add(3)
+	wg.Add(groups)
 	chanStoreError := make(chan *detailedError, 1)
 	defer close(chanStoreError)
 	chanMongoIter := make(chan mongo.StorageIterator, 1)
 	defer close(chanMongoIter)
-	chanApiError := make(chan *detailedError, 1)
-	defer close(chanApiError)
-	chanApiCbgs := make(chan []schema.CbgBucket, 1)
-	defer close(chanApiCbgs)
-	chanApiBasals := make(chan []schema.BasalBucket, 1)
-	defer close(chanApiBasals)
-	chanApiBasalError := make(chan *detailedError, 1)
-	defer close(chanApiBasalError)
 
 	// Parallel routines
 	go a.getDataFromStore(ctx, &wg, res.TraceID, params.user, dates, chanMongoIter, chanStoreError)
-	go a.getCbgFromTideV2(ctx, &wg, params.user, sessionToken, dates, chanApiCbgs, chanApiError)
-	go a.getBasalFromTideV2(ctx, &wg, params.user, sessionToken, dates, chanApiBasals, chanApiBasalError)
+
+	chanApiCbgs := make(chan []schema.CbgBucket, 1)
+	chanApiError := make(chan *detailedError, 1)
+	if params.source["cbgBucket"] {
+		fmt.Printf("cbg bucket")
+		defer close(chanApiError)
+		defer close(chanApiCbgs)
```

Comment:
defers instructions should always be declared just after the make, not sure it's best practice to move it away from declare (for the moment just few lines above, but if someone is creating a function later it can move far away from declaration)